platform/source/CThunkBase.h

```diff
@@ -21,6 +21,13 @@
 /* IRQ/Exception compatible thunk entry function */
 typedef void (*CThunkEntry)(void);
 
+/** \ingroup platform */
+/** \addtogroup platform-internal-api */
```

Comment:
class CThunk publicly derives from class CThunkBase so CThunkBase should be a public API.

Reply:
Kevin's table says Private:
https://jira.arm.com/browse/IOTCORE-1260?focusedCommentId=4118474&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-4118474

Reply:
https://github.com/ARMmbed/mbed-os/pull/11033/commits/2415d0e9c8477df2af52bed3d13c99c3de6368ad

Reply:
But there isn't anything public in CThunkBase - it's a public base, but all members are protected/private.

The definition could be changed to `class CThunk: protected CThunkBase` with no effect at all, and it probably should have been `class CThunk : private CThunkBase` - the protected members are intended only for `CThunk`'s use.

Reply:
d9b5a6c

Reply:
Thanks, but assuming we're still following the model previously discussed, anything included from a public header file should be outside the `source` directory, so it should be `platform/internal/CThunkBase.h`

Reply:
ecb9929